WindowsDevicePortalWrapper/WindowsDevicePortalWrapper.Shared/Core/DumpCollection.cs

```diff
@@ -32,23 +33,207 @@ public partial class DevicePortal
+            return dfl.DumpFiles;
+        }
 
         /// <summary>
-        /// API to retrieve a live dump from a running user mode process.
+        /// Download a dumpfile from the system. 
         /// </summary>
-        public static readonly string LiveProcessDumpApi = "api/debug/dump/usermode/live";
+        /// <param name="crashdump"> Dumpfile object to be downloaded</param>
+        /// <returns>Stream of the dump file </returns>
+        public async Task<Stream> GetDumpFileAsync(Dumpfile crashdump)
+        {
+            string queryString = BugcheckFileApi + string.Format("?filename={0}", crashdump.Filename);
+            Uri uri = Utilities.BuildEndpoint(
+                this.deviceConnection.Connection,
+                queryString);
+
+            return await this.GetAsync(uri);
+        }
+
+        /// <summary>
+        /// Takes a live kernel dump of the device.  This does not reboot the device. 
+        /// </summary>
+        /// <returns>Stream of the kernel dump</returns>
+        public async Task<Stream> GetLiveKernelDumpAsync()
+        {
+            Uri uri = Utilities.BuildEndpoint(
+                this.deviceConnection.Connection,
+                LiveKernelDumpApi);
+
+            return await this.GetAsync(uri);
+        }
+
+        /// <summary>
+        /// Take a live dump of a process on the system. 
+        /// </summary>
+        /// <param name="pid"> PID of the process to get a live dump of.</param>
+        /// <returns>Stream of the process live dump</returns>
+        public async Task<Stream> GetLiveProcessDumpAsync(int pid)
+        {
+            string queryString = LiveProcessDumpApi + string.Format("?pid={0}", pid);
+            Uri uri = Utilities.BuildEndpoint(
+                this.deviceConnection.Connection,
+                queryString);
+
+            return await this.GetAsync(uri);
+        }
+
+        /// <summary>
+        /// Get information on how and when dump files are saved on the device. 
+        /// </summary>
+        /// <returns>Dumpfile settings object.  This can be edited and returned to the device to alter the settings.</returns>
+        public async Task<DumpFileSettings> GetDumpFileSettingsAsync()
+        {
+            return await this.GetAsync<DumpFileSettings>(BugcheckSettingsApi);
+        }
+
+        /// <summary>
+        /// Set how and when dump files are saved on the device. 
+        /// </summary>
+        /// <param name="dfs">Altered DumpFileSettings object to set on the device</param>
+        /// <returns>Task tracking completion of the request</returns>
+        public async Task SetDumpFileSettingsAsync(DumpFileSettings dfs)
+        {
+            await this.PostAsync(
+                BugcheckSettingsApi,
+                string.Format(
+                    "autoreboot={0}&overwrite={1}&dumptype={2}&maxdumpcount={3}",
+                    dfs.AutoReboot ? "1" : "0", dfs.Overwrite ? "1" : "0", (int)dfs.DumpType, dfs.MaxDumpCount));
```

Comment:
There is a Utilities.BuildQueryString in the project that you can use to format this stuff (takes a dictionary of name/value pairs and formats them with the & and = signs). This seems clear enough though, so I'd be fine either way.